controllers/databaseengine_controller.go

```diff
@@ -86,55 +104,164 @@ func (r *DatabaseEngineReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 			return ctrl.Result{}, err
 		}
 	}
+
 	if version != "" {
 		dbEngine.Status.OperatorVersion = version
 		dbEngine.Status.State = everestv1alpha1.DBEngineStateInstalling
-	}
-	if ready { //nolint:nestif
-		dbEngine.Status.State = everestv1alpha1.DBEngineStateInstalled
-		matrix, err := r.versionService.GetVersions(engineType, dbEngine.Status.OperatorVersion)
-		if err != nil {
+
+		// Handle operator upgrade, if needed.
+		if done, err := r.handleOperatorUpgrade(ctx, dbEngine, ready); err != nil {
 			return ctrl.Result{}, err
```

Comment:
Would this block further dbEngine reconciliation in case of e.g. `errInstallPlanNotFound`?

Reply:
Hmm, that's a good point.

I've fixed it now so that we fallthrough in case of this error.